crates/bevy_asset/src/event.rs

```diff
@@ -11,6 +11,8 @@ pub enum AssetEvent<A: Asset> {
     Modified { id: AssetId<A> },
     /// Emitted whenever an [`Asset`] is removed.
     Removed { id: AssetId<A> },
+    /// Emitted when the last [`super::Handle::Strong`] of an [`Asset`] is dropped.
+    NoLongerUsed { id: AssetId<A> },
```

Comment:
Silly aesthetic thing: I think `NoLongerUsed` could be changed to `Unused`. The other events refer to entering a new state, not previous states. Similarly `NoLongerUsed` defines the state as "this was used in the past but now it is not". I think references to past state are spurious here. That is implied by previous events indicating that it was used previously. Also in the other states, we don't refer to other aspects of the timeline (or the timeline at all). Ex: it is `Modified` not `JustModified`. The "just" timeline indicator is implied by nature of receiving it at this time.